compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -124,6 +124,13 @@ void sym_wquant_per_channel(CircleConst *node, std::vector<float> &min, std::vec
 
   auto quantize = [&](uint32_t *indices, loco::TensorShape &dimension, int channel_dim_index) {
     int channel_idx = indices[channel_dim_index];
+
+    // Small scale factor produced by small weights, which can be quantized to 0 with scale 1.0f
+    if (scaling_factor[channel_idx] < std::numeric_limits<float>::epsilon())
+    {
+      scaling_factor[channel_idx] = 1.0f;
```

Comment:
Scale equal to 1.0 with small weight values produce zero in code bellow:

https://github.com/Samsung/ONE/blob/898ca6995b5b2b46fa074a8f098f6725327a6f39/compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp#L138

Reply:
`scaling_factor_inv == 1.0f` for `scaling_factor[channel_idx] < std::numeric_limits<float>::epsilon()`.
So `data * 1.0f` is `data`. Is this what you had in mind?

Reply:
I also find this design confusing.
If you want to quantize small small weights to 0, it makes sense to do it below, like
```c++
    if (scaling_factor < epsilon)
    {
      quantized_values[cal_offset(dimension, indices)] =
        static_cast<int32_t>(std::round(data * scaling_factor_inv));
    }
```

Reply:
> So `data * 1.0f` is `data`. Is this what you had in mind?

Yes, I mean `data * scale = data` in cases of very low scales, but very low scales can be produced only by low weights. Therefore `round(inv_scale * data) == round(data) == 0`. =)

Reply:
> I have concerns that messing with scales could lead to some unexpected side effects.

Low scales of weights produces very low scales of biases and quantized biases may have large quantization error (details: https://github.com/Samsung/ONE/issues/6970#issue-916328038)
This a reason to change scale here =)